Review

MCPN opened the pull request:
https://habr.com/ru/post/150732/

MCPN committed:
Add files via upload

MCPN committed:
Minor changes + comments

MCPN committed:
Comments

glebshp commented on avl_tree.cpp:
```diff
@@ -0,0 +1,263 @@
+            node->left = del(node->left, elem);
+        } else if (node->val < elem) {
+            node->right = del(node->right, elem);
+        } else {
```
Что происходит в этом else, нужно пояснение

glebshp commented on avl_tree.cpp:
```diff
@@ -0,0 +1,263 @@
+        return balance(node);
+    }
+
+    // Functions that used in deletion in order to swap min and elem
```
Этот комментарий:
1. Непонятен
2. Неграмотен

glebshp commented on avl_tree.cpp:
```diff
@@ -0,0 +1,263 @@
+    }
+
+    // Main AVL function: fix tree's balance
+    Node* balance(Node *node) {
```
Комментарии в этом методе достаточно бесполезные, они поясняют действия, которые и так следуют из названия функций, и не поясняют нетривиальные. Что значит баланс 2? Что значит баланс -2?

glebshp commented on avl_tree.cpp:
```diff
@@ -0,0 +1,263 @@
+        }
+    }
+
+    Set(const Set &other): Set() {
```
В данном случае принято писать const Set& other, аналогично почти во всём остальном коде. Аналогично со *

glebshp reviewed (changes requested):
Улучшить комментарии, сделать указание типов int *a -> int* a и int &a -> int& a

MCPN commented:
Произошел маркдаун (int a -> int a, звездочки заменились на курсив)

MCPN committed:
Changes requested by reviewer

1) * and & are now in type, not in variable
2) Old comments fixed
3) Add more comments
